src/core/ddsc/src/dds_write.c

```diff
@@ -99,7 +99,7 @@ static dds_return_t deliver_locally (struct writer *wr, struct ddsi_serdata *pay
     {
       dds_duration_t max_block_ms = wr->xqos->reliability.max_blocking_time;
       struct ddsi_writer_info pwr_info;
-      ddsi_make_writer_info (&pwr_info, &wr->e, wr->xqos);
+      ddsi_make_writer_info (&pwr_info, &wr->e, wr->xqos, now_mt(), payload->statusinfo);
```

Comment:
Considering that all paths simply call ``now_mt()`` to get the new 4th parameter and that it is entirely superfluous when ``DDSI_INCLUDE_LIFESPAN`` is not set, it would probably make sense to make that call inside ``ddsi_make_writer_info`` instead.

(The refactoring suggested at the beginning of ``whc_default_insert_seq`` might well affect this.)